### DISCUSSION TITLE
Recurring coupon (or discount)

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [OpenMage/magento-lts#3914] [Questions & Answers]
Looks like magento does not accept coupons for recurring products. I would like to give some users a discount on taking a monthly/yearly subscription. Thought about doing this via coupons, but unfortunately is not implemented.

Any ideas around this? (Don't think trial option is interesting, as that is for all orders)

**Comment by jparker1986** [comment]
Could you not setup a new product for yearly subscription with added discount level?

**Reply by f1-outsourcing** [reply]
You mean add a 'special price' at prices on the product page? 

I don't really know yet how much I would like to give. During a phone call I will probably offer 10%,20%,30%,40% off (brainstorming here). I have maybe ~10 different subscription models. So creating new products for all these possibilities is quite messy.

**Reply by jparker1986** [reply]
Is that no different to creating a coupon code for every discount level? Or was you intending on giving coupon code with set level of discount to the customer?

**Reply by f1-outsourcing** [reply]
If I am understanding you correctly.  I have now thought of 5 subscription services so with 4 discount rates. I would end up with adding 9 'entries' to magento. 
If I would have to make a new product for every rate. I would end up with 5 (original) + 5 x 4 (discount) is 25 entries. I think this would start to look messy.

**Reply by f1-outsourcing** [reply]
Yesterday I was a bit searching in the code if there is somewhere a function that checks the product on having recurring, and try to comment that out. Just see what happens. 

I assume they removed this because of not being able to control fully the future recurring orders. But I just need this for the first.

**Reply by jparker1986** [reply]
It already becomes messy if your offering the same product with multiple discount rates for the same subscription length. 

Could you not instead assign the discount level to the customer group and assign a customer to the group based on what level of discount you want to offer?

**Reply by f1-outsourcing** [reply]
True, but I am only giving the discount the first period, not the recurring. Sort of an onboarding incentive.

This customer group is indeed interesting, I could do this temporary, and remove them before next billing cycle. Thanks! :)

**Reply by f1-outsourcing** [reply]
This is not possible / bugs.
1. In order to use customer classes I need to create the customer in the admin (grrr), assign the class. However I can't create order for recurring product (bugs: order totals stays 0, whining about shipping while it is virtual etc)
2. Using the created user account to login in on the front end with one page check out, does not apply the this discount.

**Reply by f1-outsourcing** [reply]
I think coupons should work as I initially wanted to use them. I think there is a bug in calculating totals, which prevents the coupon to be applied
https://github.com/OpenMage/magento-lts/issues/3916
